recursion/reading-recursive-code.md

```diff
@@ -0,0 +1,288 @@
+| Recursion (computer science) | A programming technique where a function calls itself                                                                                                                 | "One way to compute the nth fibonacci number is with recursion, where a function `fibonacci` calls itself until it's finished."                        |
+| Base Case                    | A condition which will end the recursion. This is case where the solution is concrete                                                                                 | "The base case for computing the nth fibonacci number is when `n` is 1 or 2, as the first two numbers of the fibonacci sequence are always 1 and 1." " |
+| Recursive Case               | A situation in a recursive function that will call the function itself                                                                                                | "When computing the nth fibonacci number, if `n` is greater than 2, the `fibonacci` function will call itself."                                        |
+| Stack (data structure)       | A data structure that operates as "Last-In-First-Out," where new data is pushed onto the top, and data is popped off from the top                                     |
+| System stack                 | A part of memory dedicated to storing the method calls and local variables of a running program. The system stack is responsible for keeping track of function calls. |
+
+## Recursion
+
+In programming, a recursive function is a function that calls itself.
+
+We can imagine a recursive function named `get_next_item`. When we call `get_next_item`, it will invoke itself, `get_next_item`. Of course, when _this_ function calls begins execution, it will also invoke itself.
+
+Both iteration and recursion are programming techniques used to repeat logic and behavior. There are often iterative and recursive solutions to the same problem!
+
+When would we choose to solve a problem with recursion? Once we're used to thinking about recursion, recursive algorithms may be more readable and understandable compared to iterative solutions.
+
+## Anatomy of a Recursive Function
+
+Functions that call itself, like the imaginary `get_next_item` function, will recurse an infinite number of times.
+
+However, if a function calls itself an infinite number of times, when will we actually get an answer and solve our problem? Recursive functions are more helpful when they actually _do_ stop!
+
+In order to better read and debug recursive functions, let's dissect the anatomy of one.
+
+### Base Cases are Stopping Conditions
+
+Recursive algorithms will usually take in at least one argument. A recursive algorithm handles different situations based on the value of the argument. The algorithm should handle:
+
+- At least one _recursive case_
+- At least one _base case_
```

Comment:
reverse these since usually we check for the base case first, then fall back to the recursive case
```suggestion
- At least one _base case_
- At least one _recursive case_
```